supabase_auth/_sync/gotrue_base_api.py

```diff
@@ -97,6 +98,8 @@ def _request(
     ) -> Union[T, None]:
         url = f"{self._url}/{path}"
         headers = {**self._headers, **(headers or {})}
+        if headers.get(API_VERSION_HEADER_NAME) is None:
```

Comment:
This looks incorrect. If there's no `API_VERSION_HEADER_NAME` the server is probably on an old version (or potentially CORS misconfigured). In that case the default shouldn't be 2024-01-01 but rather the _initial_ unspecified API version.

You can represent this as `1970-01-01` or something equivalent to it. None is fine too.

Reply:
This is the equivalent of what's in the JS library https://github.com/supabase/auth-js/blob/master/src/lib/fetch.ts#L137-L139, so is the JS library code incorrect?